src/csm_testing/lib/iuf_classes.py

```diff
@@ -0,0 +1,257 @@
+        self.resource = os.getenv("IUF_API_URL_RESOURCE", resource)
+
+    def request(self, method, path, payload=None, timeout=None):
+        method = method.upper()
+        assert method in ['GET', 'HEAD', 'DELETE', 'POST', 'PUT',
+                          'PATCH', 'OPTIONS']
+
+        url=self.apiurl + self.resource + path
+
+        headers = dict()
+        try:
+            token = None
+            headers["Authorization"] = f"Bearer {token}"
+        except:
+            if "gw-service" in self.apiurl:
+                raise
+            else:
+                # if we're not using the "official" api and don't get a token just try without it.  Mostly for local testing.
+                pass
+
+        method_func = method.lower()
+        try:
+            if payload:
+                result = getattr(requests, method_func)(url, headers=headers, json=payload, verify=False, timeout=timeout)
+            else:
+                result = getattr(requests, method_func)(url, headers=headers, verify=False, timeout=timeout)
+        except:
+            raise
+
+        # throw an exception for bad status codes
+        result.raise_for_status()
+
+        return result
+
+    def get_stages(self):
+        api_path = f"/stages"
+        try:
+            api_response = self.request("GET", api_path)
+            return api_response
+        except:
+            raise
+
+class ApiInterface(object):
+    def __init__(self, apiurl="https://api-gw-service-nmn.local/apis", resource="/iuf/v1"):
+        self.auth = Auth()
+        self.apiurl = os.getenv("IUF_API_URL", apiurl)
+        self.resource = os.getenv("IUF_API_URL_RESOURCE", resource)
+
+    def request(self, method, path, payload=None, timeout=None):
+        method = method.upper()
+        assert method in ['GET', 'HEAD', 'DELETE', 'POST', 'PUT',
+                          'PATCH', 'OPTIONS']
+
+        url=self.apiurl + self.resource + path
+
+        headers = dict()
+        try:
+            token = self.auth.token
+            headers["Authorization"] = f"Bearer {token}"
+        except:
+            if "gw-service" in self.apiurl:
+                raise
+            else:
+                # if we're not using the "official" api and don't get a token just try without it.  Mostly for local testing.
+                pass
+
+        method_func = method.lower()
+        try:
+            if payload:
+                result = getattr(requests, method_func)(url, headers=headers, json=payload, verify=False, timeout=timeout)
+            else:
+                result = getattr(requests, method_func)(url, headers=headers, verify=False, timeout=timeout)
+        except:
+            raise
+
+        # throw an exception for bad status codes
+        result.raise_for_status()
+
+        return result
+    
+    def activity_exists(self, activity):
+        try:
+            self.get_activity(activity)
+            return True
+        except:
```

Comment:
Bare `except` statements should be avoided. They catch everything, including fatal system-level exceptions, which is not usually a good thing to do.

Ideally, all except statements should specify the types of exceptions they are going to catch. But in cases where that is not feasible, it should at least just catch `Exception`. That will not blindly catch super-low-level exceptions.